src/plugins/layout/shadowdom.ts

```diff
@@ -347,6 +350,22 @@ export class ShadowDom {
     return parentShadowNode && !this.hasClass(parentShadowNode, FinalClassName);
   }
 
+  // When we apply a mutation batch, we assign a new index to every node that is added to the DOM as we parse mutations
+  // in their natural order. However, some of those nodes end up being again removed from DOM with the some follow-up mutation
+  // thus 'stealing' an index with them. Since we don't instrument nodes that were added and removed within the same
+  // mutation batch, we have no records of these stolen indices. To maintain consistency on the backend, at the end of the mutation,
+  // we can re-assign indices for all nodes that remained attached to the DOM such that they all go in increasing order without gaps
+  private reIndexNewNodes(nextIndex: number) {
+    let newNodes = this.doc.getElementsByClassName(NewNodeClassName);
+    for (let i = 0; i < newNodes.length; i++) {
+      let shadowNode = newNodes[i] as IShadowDomNode;
```

Comment:
This can't be done, because newNodes is an HTMLCollection, which cannot be converted to an array.

---
In reply to: [131240755](https://github.com/Microsoft/clarity-js/pull/25#discussion_r131240755) [](ancestors = 131240755)